src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.String.cs

```diff
@@ -103,6 +114,22 @@ public bool TryReadTextString(Span<char> destination, out int charsWritten)
             return true;
         }
 
+        // SkipValue() helper method, reads a text string without allocating/copying to a buffer
+        // NB this only handles definite-length chunks
+        private void SkipTextString()
+        {
+            CborInitialByte header = PeekInitialByte(expectedType: CborMajorType.TextString);
+            int byteLength = checked((int)ReadUnsignedInteger(_buffer.Span, header, out int additionalBytes));
+            EnsureBuffer(1 + additionalBytes + byteLength);
+
+            // force decoding errors if binary is not valid utf8
+            ReadOnlySpan<byte> encodedSlice = _buffer.Span.Slice(1 + additionalBytes, byteLength);
+            s_utf8Encoding.GetCharCount(encodedSlice);
```

Comment:
I'd probably leave out the encoding validation.  If you're skipping over the string tagged as a greeting, was it important that it was valid UTF-8?

Then the two SkipString methods can be merged into one that just takes the expectedType.

Reply:
> was it important that it was valid UTF-8?

I think it's important that it behaves consistently from a format validation perspective, regardless of whether we're reading or skipping over a value. Among other things, the method should provide a cheap way to validate whether the data we're reading is valid CBOR.